src/main/java/com/avioconsulting/mule/logger/internal/listeners/CustomLoggerAbstractNotificationListener.java

```diff
@@ -41,4 +47,56 @@ protected void logMessage(ComponentLocation location, Event event, String logMes
     logger.log(logProperties, messageAttributes, exceptionProperties, additionalProperties, config,
         location, correlationId);
   }
-}
+
+  protected Map<String, String> getFlowLogAttributes(EnrichedServerNotification notification) {
+    Map<String, String> value = emptyAttributes;
+    FlowLogConfig flowLogConfig;
+    /**
+     * Flow name can contain wildcard (*)
+     * We only look for wildcard either starting of the string or ending of the
+     * string
+     * ex: mq-listener-* will look for all the flows that starts with mq-listener
+     * ex: *-mq-flow will look for all the flows that ends with -mq-flow
+     **/
+    List<Map.Entry<String, FlowLogConfig>> matchedEntries = config.getFlowLogConfigMap().entrySet().stream()
+        .filter(entry -> matchWildcard(entry.getKey(), notification.getResourceIdentifier()))
+        .collect(Collectors.toList());
+    if (!matchedEntries.isEmpty()) {
+      flowLogConfig = matchedEntries.get(0).getValue();
+      TypedValue<Map<String, String>> evaluate = (TypedValue<Map<String, String>>) config
+          .getExpressionManager()
+          .evaluate("#[" + flowLogConfig.getExpressionText().getAttributesExpressionText() + "]",
+              notification.getEvent().asBindingContext());
+      value = evaluate.getValue();
+      if (value == null)
+        value = emptyAttributes;
+    }
+    return value;
+  }
+
+  public boolean matchWildcard(String wildcardKey, String searchString) {
+    // Trim the wildcard key
+    String cleanWildcardKey = wildcardKey.trim();
+
+    // Exact match if no wildcards
+    if (searchString.equals(wildcardKey)) {
```

Comment:
Shouldn't this be a case insensitive comparison?

Reply:
updated to use "equalsIgnoreCase" and "toLowerCase"

Reply:
Sorry, one last thing - use of both is redundant -
```
if (searchString.equalsIgnoreCase(wildcardKey.toLowerCase())) {
```

should be just this -
```
if (searchString.equalsIgnoreCase(wildcardKey)) {
```

Reply:
no worries, just updated